pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -299,47 +313,12 @@ func (vs *VSphere) getVMByName(ctx context.Context, nodeName k8stypes.NodeName)
 // NodeAddresses is an implementation of Instances.NodeAddresses.
 func (vs *VSphere) NodeAddresses(nodeName k8stypes.NodeName) ([]v1.NodeAddress, error) {
 	// Get local IP addresses if node is local node
-	if vs.localInstanceID == nodeNameToVMName(nodeName) {
+	if vs.hostName == nodeNameToVMName(nodeName) {
 		return getLocalIP()
 	}
-	addrs := []v1.NodeAddress{}
```

Comment:
Why is the code block L305:L342 removed?
What is the output of `kubectl describe node node1 | grep Address`
```
$ kubectl describe node node1 | grep Address
Addresses:		10.162.12.189,10.162.12.189,node1
```
This should not return internal addresses and ipv6 addresses